qiskit/primitives/base/base_estimator.py

```diff
@@ -282,6 +282,20 @@ def _cross_validate_circuits_observables(
                     f"({observable.num_qubits})."
                 )
 
+    @classmethod
+    def _validate_circuits(
+        cls,
+        circuits: Sequence[QuantumCircuit] | QuantumCircuit,
+    ) -> tuple[QuantumCircuit, ...]:
+        circuits = super()._validate_circuits(circuits)
+        for i, circuit in enumerate(circuits):
+            if circuit.num_clbits > 0:
+                raise ValueError(
+                    f"The {i}-th circuit has some classical bits. "
+                    "Estimator accepts quantum circuits without classical bits."
+                )
```

Comment:
Yes, you are right. Idle clbits do not affect estimator results. But, backend estimator is designed based on the original primitive spec that the input circuit has no clbits. So, it currently raises an error with such a circuit because it uses clbits to identify qubit mapping (as well as Runtime Estimator). On the other hand, the "reference" estimator based on `Statevector` can handle such circuits.

It might be a good opportunity to update the backend estimator to allow idle clbits.

```python
from qiskit.primitives import Estimator as RefEstimator
from qiskit.primitives import BackendEstimator
from qiskit_aer import AerSimulator
from qiskit import QuantumCircuit
from qiskit.quantum_info import SparsePauliOp

qc = QuantumCircuit(2, 2)
op = SparsePauliOp("II")

estimator = RefEstimator()
result = estimator.run([qc], [op]).result()  # OK
print(result)

backend = AerSimulator()
estimator = BackendEstimator(backend)
result = estimator.run([qc], [op]).result()  # Error: qiskit.circuit.exceptions.CircuitError: 'register name "c" already exists'
print(result)
```